system/src/main.cpp

```diff
@@ -65,52 +93,118 @@ static volatile uint32_t TimingIWDGReload;
 {
     if (button || network.listening())
         return 0;
-    return buttonPushed ? HAL_Timer_Get_Milli_Seconds()-buttonPushed : 0;
+    return pressed_time ? HAL_Timer_Get_Milli_Seconds()-pressed_time : 0;
 }
 
+#if Wiring_SetupButtonUX
+/* flag used to initiate system_handle_single_click() from main thread */
+static volatile bool SYSTEM_HANDLE_SINGLE_CLICK = false;
+
 static uint32_t prev_release_time = 0;
 static uint8_t clicks = 0;
 
+/* single click handler displays RSSI value on system LED */
+void system_handle_single_click()
+{
+    /*   signal strength (u-Blox Sara U2 and G3 modules)
+     *   0: < -105 dBm
+     *   1: < -93 dBm
+     *   2: < -81 dBm
+     *   3: < -69 dBm
+     *   4: < - 57 dBm
+     *   5: >= -57 dBm
+     */
+    if (SYSTEM_HANDLE_SINGLE_CLICK) {
+        SYSTEM_HANDLE_SINGLE_CLICK = false;
+        volatile system_tick_t startTime = SYSTEM_TICK_COUNTER;
+        RGB.control(true);
+        RGB.color(0,10,0);
+        int rssi = 0;
+        int bars = 0;
+#if Wiring_WiFi == 1
+        rssi = WiFi.RSSI();
+#elif Wiring_Cellular == 1
+        CellularSignal sig = Cellular.RSSI();
+        rssi = sig.rssi;
+#endif
+        if (rssi < 0) {
+            if (rssi >= -57) bars = 5;
+            else if (rssi > -68) bars = 4;
+            else if (rssi > -80) bars = 3;
+            else if (rssi > -92) bars = 2;
+            else if (rssi > -104) bars = 1;
+        }
+        DEBUG("RSSI: %ddB BARS: %d\r\n", rssi, bars);
+
+        /* flash sequence */
+        /* TODO - REFACTOR Flash Sequence to be non-blocking via HAL_SysTick_Handler() */
+
+        // Attempts to normalize beginning dim green time to 1 second
+        while ( (SYSTEM_TICK_COUNTER - startTime) < (SYSTEM_US_TICKS*1000UL*1000UL) );
+
+        if (bars > 0) {
+            for (int x=0; x<bars; x++) {
+                RGB.color(0,255,0);
+                delay(40);
+                RGB.color(0,10,0);
+                delay(250);
+            }
+        }
+        delay(750);
+        RGB.control(false);
+    }
+}
+
 void system_handle_double_click()
 {
     SYSTEM_POWEROFF = 1;
     network.connect_cancel(true, true);
 }
 
-void handle_button_click(uint16_t duration, uint32_t release_time)
+void reset_button_click(uint32_t release_time)
 {
-    uint32_t start_time = release_time-duration;
-    uint32_t since_prev = start_time-prev_release_time;
+    clicks = 0;
+    prev_release_time = release_time - 1000;
+    CLR_BUTTON_TIMEOUT();
+}
+
+void handle_button_click(uint16_t depressed_duration, uint32_t release_time)
+{
+    uint32_t start_time = release_time - depressed_duration;
+    uint32_t since_prev = start_time - prev_release_time;
     bool reset = true;
-    if (duration<500) {                                 // a short button press
+    if (depressed_duration<500) {               // a short button press
         if (!clicks || (since_prev<1000)) {		// first click or within 1 second of the previous click
```

Comment:
So we give the user 1s in between clicks to trigger the RSSI mode? Seems long enough, but was this an arbitrary choice?